doc/modules/grid_search.rst

```diff
@@ -192,11 +192,53 @@ result in an error when using multiple metrics.
 See :ref:`sphx_glr_auto_examples_model_selection_plot_multi_metric_evaluation.py`
 for an example usage.
 
+.. _composite_grid_search:
+
 Composite estimators and parameter spaces
 -----------------------------------------
+`GridSearchCV` and `RandomizedSearchCV` allow searching over parameters of
+composite or nested estimators such as `Pipeline`, `ColumnTransformer`,
+`VotingClasssifier` or `CalibratedClassifierCV`
+using a dedicated syntax ``<estimator>__<parameter>`` syntax::
+
+  >>> from sklearn.model_selection import GridSearchCV
+  >>> from sklearn.calibration import CalibratedClassifierCV
+  >>> from sklearn.ensemble import RandomForestClassifier
+  >>> from sklearn.datasets import make_moons
+  >>> X, y = make_moons()
+  >>> calibrated_forest = CalibratedClassifierCV(
+  ...    base_estimator=RandomForestClassifier(n_estimators=10))
+  >>> param_grid = {
+  ...    'base_estimator__max_depth': [2, 4, 6, 8]}
+  >>> search = GridSearchCV(calibrated_forest, param_grid, cv=5)
+  >>> search.fit(X, y)
+  GridSearchCV(cv=5,
+               estimator=CalibratedClassifierCV(base_estimator=RandomForestClassifier(n_estimators=10)),
+               param_grid={'base_estimator__max_depth': [2, 4, 6, 8]})
+
+Here, ``<estimator>`` is the parameter name of the nested estimator,
+in this case ``base_estimator``.
+The `Pipeline` class has a slightly different notation, as explained in :ref:`pipeline`,
+where ``<estimators>`` referes to the name of the step in the pipeline.
+In practice, there can be several levels of nesting::
+
+  >>> from sklearn.pipeline import Pipeline
+  >>> from sklearn.feature_selection import SelectKBest
+  >>> pipe = Pipeline([
+  ...    ('select', SelectKBest()),
+  ...    ('model', calibrated_forest)])
+  >>> param_grid = {
+  ...    'select__k': [1, 2],
+  ...    'model__base_estimator__max_depth': [2, 4, 6, 8]}
+  >>> search = GridSearchCV(pipe, param_grid, cv=5)
+  >>> search.fit(X, y)
+  GridSearchCV(cv=5,
+               estimator=Pipeline(steps=[('select', SelectKBest()),
```

Comment:
This is quite wide. Does doing something like this hide too much?

```py
GridSearchCV(cv=5,
             estimator=Pipeline(steps=[...]),
             param_grid={'model__base_estimator__max_depth': [2, 4, 6, 8],
                         'select__k': [1, 2]})
```

Reply:
it would be nice to format this differently but pytests doesn't allow me to.
We could change our repr lol.
I can also hide more with ellipsis but I think that's not super nice.

Reply:
IGNORE_WHITESPACE doesn't allow changes in line breaks?? Who knew?

More practically, maybe just fit on the previous line and don't show the repr?

Reply:
it allows replacing any whitespace by any other whitespace. There's no whitespace here to replace :-/